features/FxA-106-signin-unblock/README.md

```diff
@@ -0,0 +1,285 @@
+
+When our fraud-and-abuse system generates a false negative and allows a malicious login attempt with a guessed or breached password, we have to rely on the sign-in confirmation feature as additional security for the user's sync data. The attacker still learns whether their login attempt succeeded, meaning they still learn whether they have the correct account password and can still access other FxA relying services.
+
+With this feature, we aim to lower the impact of false positives when blocking login attempts, by providing clearer messaging to the user about what's happening and why, and by allowing legitimate users to unblock their requests by proving ownership of the account. These measures will give us more confidence to tighten the security rules and thus reduce the rate of false negatives, improving overall security of the system.
+
+## Outcomes
+
+Users whose login attempts are blocked by our fraud-and-abuse system will be able to unblock themselves by doing an email verification loop to prove ownership of the account. As a result we will be able to:
+
+* Implement stricter rules to detect fraudulent access, and simultaneously
+* Decrease the number of legitimate login flows abandoned due to security measures.
+
+## Hypothesis
+
+If we send emails that allow users to unblock themselves after being flagged as suspicious by security measures, then we will increase the completion rate of legitimate login attempts without allowing significantly more fraudulent logins, because users will be able to immediately continue without waiting for a time period to elapse.
+
+We will know this to be true when we see an increase in the completion rate of login attempts that triggered a security measures, as a percentage of all login attempts
+that trigger a security measure.
+Metrics
+
+As a high-level success metric, we want to know the following:
+
+> Of all login attempts that were blocked a security measure, what percentage are subsequently completed successfully?
+
+We expect the starting value of this metric to be close to zero, since it is not obvious to users whether and how they can complete the flow after being blocked by a security measure. They currently can, in some cases, by waiting for a rate-limiting ban to expire, but this is not explained in the UI and users are unlikely to do so in practice.
+
+TODO: mock graph here
+
+As a detailed metric of the effectiveness of the technical details of this feature, we want to know how many users...:
+
+* Had login requests blocked for security reasons?
+* Chose to verify their account to unblock themselves?
+* Successfully logged in after verifying ownership of the account?
+* Chose to report the login?
+
+Taken together, these metrics form a success funnel for the email authorization feature.
+
+TODO: mock graph here
```

Comment:
We need to make sure we know what events will be used to produce this graph, and that we're actually emitting them.  Are we using flow events for this?  Is so then I imagine the lines of the graph being:

* Number of distinct flows that contain an `account.login.blocked` event
* Number of distinct flows that contain an `account.login.sentUnblockCode` event
* Number of distinct flows that contain an `account.login.confirmedUnblockCode` event
* Number of distinct flows that contain an `account.login.rejectedUnblockCode` event

Also I made a quick mock:
![mock graph](https://cloud.githubusercontent.com/assets/34695/19460634/19ae4aba-9529-11e6-9f88-85feb1f66050.jpg)